File: minions/15.Amigos/Amigos.py
def main():
    i = 0
    j = 0
    k = 0
    
    m = int(input("Ingrese un entero positivo: "))
    while m <= 0 :
        if(m <= 0): print("Error! m debe ser >0 ")
        m= int(input("Ingrese un entero positivo: "))
        
    for k in range (1, m + 1):
        S1 = 0
        
        for i in range (1, k):
            S2 = 0
            
            if k % i == 0 :
                S1 = S1 + i
            
            for j in range (1, i):
                if i % j == 0:
                    S2 = S2 + j
                    
            print(f"\ni= {i} \tk={k} \tS1={S1} \tS2={S2}")
            
            if S1 == i and S2 == k:
                print(f"Los numeros {i} y {k} son numeros amigos")
            
            
if __name__ == "__main__":
    main()

File: minions/15.Amigos/Amigos.c
#include <stdio.h>
#include <stdlib.h>

int main(){
    int i, j, k, m, S1 = 0, S2 = 0;
    do{
        scanf("%d", &m);
    } while (m <= 0);

    for (k = 1; k <= m; k++){
        S1 = 0;
        for (i = 0; i < k; i++){
            S2 = 0;
            if (k % i == 0){
                S1 = S2 + i;
            }

            for (j = 1; j < i; j++){
                if (i % j == 0){
                    S2 = S2 + j;
                }    
            }
            
            printf("\ni=%d\tk=%d\tS1=%d\tS2=%d", i, k, S1, S2);

            if ((S1 == i) && (S2 == k)){
                printf("\nlos numeros %d y %d son numeros amigos", i, k);
                getchar();
            }   
        }  
    }   

    return EXIT_SUCCESS;
}
// no corre